src/test/kotlin/org/rustSlowTests/lang/resolve/CargoProjectResolveTest.kt

```diff
@@ -947,6 +947,79 @@ class CargoProjectResolveTest : RsWithToolchainTestBase() {
         prj.checkReferenceIsResolved<RsPath>("project_2/src/main.rs")
     }
 
+    fun `test cargo features namespaced dependencies and weak dependency features`() = buildProject {
+        toml("Cargo.toml", """
+            [package]
+            name = "hello"
+            version = "0.1.0"
+
+            [features]
+            feature_foo = ["dep:foo"]
+            feature_bar = ["dep:bar"]
+            feature_baz = ["foo?/baz"]
+
+            [dependencies]
+            foo = { path = "./foo", optional = true }
+            bar = { path = "./bar", optional = true }
+        """)
+
+        dir("src") {
+            rust("main.rs", """
+                fn main() {
+                    foo::foo_fn();
+                }       //^
+            """)
+        }
+
+        dir("foo") {
+            toml("Cargo.toml", """
+                [package]
+                name = "foo"
+                version = "0.1.0"
+
+                [features]
+                baz = []
+            """)
+
+            dir("src") {
+                rust("lib.rs", """
+                    #[cfg(feature = "baz")]
+                    pub fn foo_fn() {}
+                """)
+            }
+        }
+
+        dir("bar") {
+            toml("Cargo.toml", """
+                [package]
+                name = "bar"
+                version = "0.1.0"
+            """)
+
+            dir("src") {
+                rust("lib.rs", "pub fn bar_fn() {}")
+            }
+        }
+    }.run {
+        val pkg = project.cargoProjects.singlePackage("hello")
+        assertEquals(setOf("feature_foo", "feature_bar", "feature_baz"), pkg.featureState.keys)
+
+        pkg.checkFeatureEnabled("feature_foo")
+        pkg.checkFeatureEnabled("feature_bar")
+        pkg.checkFeatureEnabled("feature_baz")
+        project.cargoProjects.singlePackage("foo").checkFeatureEnabled("baz")
```

Comment:
[minor] `project.cargoProjects.singlePackage("foo")`  could be extracted to a variable

Reply:
It actually can't because modifying features leads to project re-creation